pkg/trace/obfuscate/sql_test.go

```diff
@@ -473,79 +532,135 @@ func TestSQLQuantizer(t *testing.T) {
 			`SELECT * FROM foo LEFT JOIN bar ON 'backslash\' = foo.b LEFT JOIN bar2 ON 'backslash2\' = foo.b2 WHERE foo.name = 'String'`,
 			"SELECT * FROM foo LEFT JOIN bar ON ? = foo.b LEFT JOIN bar2 ON ? = foo.b2 WHERE foo.name = ?",
+			"SELECT * FROM foo LEFT JOIN bar ON ? = foo.b LEFT JOIN bar2 ON ? = foo.b2 WHERE foo.name = ?",
 		},
 		{
 			`SELECT * FROM foo LEFT JOIN bar ON 'embedded ''quote'' in string' = foo.b WHERE foo.name = 'String'`,
 			"SELECT * FROM foo LEFT JOIN bar ON ? = foo.b WHERE foo.name = ?",
+			"SELECT * FROM foo LEFT JOIN bar ON ? = foo.b WHERE foo.name = ?",
 		},
 		{
 			`SELECT * FROM foo LEFT JOIN bar ON 'embedded \'quote\' in string' = foo.b WHERE foo.name = 'String'`,
 			"SELECT * FROM foo LEFT JOIN bar ON ? = foo.b WHERE foo.name = ?",
+			"SELECT * FROM foo LEFT JOIN bar ON ? = foo.b WHERE foo.name = ?",
+		},
+		{
+			"SELECT COUNT ( * ) AS `count_1` FROM ( SELECT `customer` . `id_` AS `customer_id_` FROM `customer` ) AS `anon_1`",
+			"SELECT COUNT ( * ) FROM ( SELECT customer . id_ FROM customer )",
+			"SELECT COUNT ( * ) FROM ( SELECT customer.id_ FROM customer )",
+		},
+		{
+			`DELETE FROM t1
+			WHERE s11 > ANY
+			 (SELECT COUNT(*) /* no hint */ FROM t2
+			  WHERE NOT EXISTS
+			   (SELECT * FROM t3
+				WHERE ROW(5*t2.s1,77)=
+				 (SELECT 50,11*s1 FROM t4 UNION SELECT 50,77 FROM
+				  (SELECT * FROM t5) AS t5)));`,
+			"DELETE FROM t1 WHERE s11 > ANY ( SELECT COUNT ( * ) FROM t2 WHERE NOT EXISTS ( SELECT * FROM t3 WHERE ROW ( ? * t2.s1, ? ) = ( SELECT ? * s1 FROM t4 UNION SELECT ? FROM ( SELECT * FROM t5 ) ) ) )",
+			"DELETE FROM t1 WHERE s11 > ANY ( SELECT COUNT ( * ) FROM t2 WHERE NOT EXISTS ( SELECT * FROM t3 WHERE ROW ( ? * t2.s1, ? ) = ( SELECT ? * s1 FROM t4 UNION SELECT ? FROM ( SELECT * FROM t5 ) ) ) )",
+		},
+		{
+			`WITH RECURSIVE cte AS
+			(
+			  SELECT 1 AS n, 'abc' AS str
+			  UNION ALL
+			  SELECT n + 1, CONCAT(str, str) FROM cte WHERE n < 3
+			)
+			SELECT * FROM cte;`,
+			"WITH RECURSIVE cte SELECT ?, ? UNION ALL SELECT n + ? CONCAT ( str, str ) FROM cte WHERE n < ? ) SELECT * FROM cte",
+			"WITH RECURSIVE cte ( SELECT ?, ? UNION ALL SELECT n + ? CONCAT ( str, str ) FROM cte WHERE n < ? ) SELECT * FROM cte",
+		},
+		{
+			"SELECT org_id,metric_key,metric_type,interval FROM metrics_metadata WHERE org_id = ? AND metric_key = ANY(ARRAY[?,?,?,?,?])",
+			"SELECT org_id, metric_key, metric_type, interval FROM metrics_metadata WHERE org_id = ? AND metric_key = ANY ( ARRAY [ ?, ?, ?, ?, ? ] )",
+			"SELECT org_id, metric_key, metric_type, interval FROM metrics_metadata WHERE org_id = ? AND metric_key = ANY ( ARRAY [ ? ] )",
 		},
 	}
 
 	for _, c := range cases {
-		t.Run("", func(t *testing.T) {
+		t.Run("denormalized", func(t *testing.T) {
 			s := SQLSpan(c.query)
 			NewObfuscator(nil).Obfuscate(s)
 			assert.Equal(t, c.expected, s.Resource)
 		})
+		t.Run("normalized", func(t *testing.T) {
+			s := SQLSpan(c.query)
+			o := NewObfuscator(nil)
+			o.sqlNewNormalization = true
```

Comment:
Maybe you can add a function into the `pkg/trace/test/testutil` package, something like:
```go
// EnableFeature activates feature f and returns a function which clears
// all feature flags. Example usage:
//
// 	defer EnableFeature("sql_reduce")()
func EnableFeature(f string) func() {
	env := "DD_APM_FEATURES"
	os.Setenv(env, os.Getenv(env)+",sql_reduce")
	return func() { os.Unsetenv(env) }
}
```
And then use it like this at the beginning of the test:
```go
defer testutil.EnableFeature("sql_normalize")()
```
What do you think?